lib/puppetserver/ca/action/generate.rb

```diff
@@ -230,9 +233,12 @@ def generate_certs(certnames, alt_names, settings, digest)
             # Check if the CA autosigned the cert
             if download_cert(ca, certname, settings)
```

Comment:
could we move this into a separate method? the containing method is already big and getting bigger, this conditional seems like a good chunk to split out??